exporters/otlp/CMakeLists.txt

```diff
@@ -37,12 +37,13 @@ if(BUILD_TESTING)
     TEST_LIST otlp_recordable_test)
   if(MSVC)
     add_definitions(-DGTEST_LINKED_AS_SHARED_LIBRARY=1)
-  endif()
-  if(GMOCK_LIB)
-    # unset GMOCK_LIB to force find_library to redo the lookup, as the cached
-    # entry could cause linking to incorrect flavor of gmock and leading to
-    # runtime error.
-    unset(GMOCK_LIB CACHE)
+  else()
+    if(GMOCK_LIB)
```

Comment:
@ThomsonTan - Tom, sorry, I kept it in here because I still have that issue popping up occasionally. I don't know why :(